### PR TITLE
[ADHOC] chore(evm): sweep the dust in boost core

### PR DESCRIPTION
**Pull request by topocount**
### Description

If residual token amounts accumulate due to integer division, this helper
will transfer those amounts to the protocol fee recipient inline during
a claim protocol fee settlement.

💔 Thank you!

### CODE REVIEW

File: packages/evm/contracts/BoostCore.sol [hunk 1]
```diff
@@ -572,10 +584,19 @@ contract BoostCore is Ownable, ReentrancyGuard {
     function _transferProtocolFee(IncentiveDisbursalInfo storage incentive, uint256 amount) internal {
         if (incentive.assetType == ABudget.AssetType.ERC20 || incentive.assetType == ABudget.AssetType.ETH) {
             incentive.asset.safeTransfer(protocolFeeReceiver, amount);
+            _dustHatch(incentive.asset);
```
Comment (Quazia) [inline packages/evm/contracts/BoostCore.sol:587]: Nice placement I like this

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a7a1a4f640410af8d67d3dde60af0486ea18c29e
</p>

File: packages/evm/test/BoostCore.t.sol [hunk 2]
```diff
@@ -555,6 +556,75 @@ contract BoostCoreTest is Test {
...
     }
 
+    function testFuzzClaimIncentive_ProtocolFeeTransfer(uint256 rewardAmount) public {
+        uint160 claimant = uint160(makeAddr("claimant"));
+        uint16 claimLimit = 1_000;
+        rewardAmount = bound(rewardAmount, 1 ether, (type(uint256).max >> 15) / claimLimit);
+        uint256 amountToMint =
+            (rewardAmount + rewardAmount * boostCore.protocolFee() / boostCore.FEE_DENOMINATOR()) * claimLimit;
+        mockERC20.mint(address(this), amountToMint);
+        mockERC20.mint(address(this), amountToMint);
```
Comment (Quazia) [inline packages/evm/test/BoostCore.t.sol:566]: Why double?

**Review by Quazia — APPROVED** [review]
LG

File: packages/evm/test/BoostCore.t.sol (hunk 2 above)
Comment (topocount) [reply packages/evm/test/BoostCore.t.sol:566]: lol there was no reason nice catch

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Are you sure you want to be submitting a change without including a changeset? If you're just changing docs or tests, you probably don't need to. See [the publishing section of the README](https://github.com/boostxyz/boost-protocol?tab=readme-ov-file#changesets--publishing) for more info.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ac8c66b0d2f95ee9d0d49f88f189b56fa3878112
</p>

**Pull request merged**